lib/biker.rb

```diff
@@ -1,3 +1,26 @@
 class Biker
+  attr_reader :name, :max_distance, :rides, :acceptable_terrain
 
-end
\ No newline at end of file
+  def initialize(name, max_distance)
+    @name = name
+    @max_distance = max_distance
+    @rides = Hash.new {|ride, time| hash[key] = []}
+    @acceptable_terrain = []
+  end
+
+  def learn_terrain!(terrain)
+    @acceptable_terrain << terrain
+  end
+
+  def log_ride(ride, time)
+    if @acceptable_terrain.include?(ride.terrain) && ride.total_distance <= @max_disatance
+      @rides[ride] << time
```

Comment:
Without setting up your default value in your initialize, you could do something like this:
```
if @rides.has_key?(ride)
    # ride already added to the hash, so you can do the next line no problem
   @rides[ride] << time
else
   @ride[ride] = [time] # ride key not added, so to avoid nil, we just add that time inside an array manually
end
```